universum/modules/error_state.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+import inspect
+
+from ..lib.gravity import Module, Dependency
+
+
+class GlobalErrorState(Module):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)  # type: ignore
+        self.errors: List[str] = []
+
+
+class ErrorState(Module):
+    global_error_state_factory = Dependency(GlobalErrorState)
+
+    def __init__(self, *args, **kwargs) -> None:
+        # mypy doesn't support cooperative inheritance
+        super().__init__(*args, **kwargs)  # type: ignore
+        self.global_error_state = self.global_error_state_factory()
+
+    def error(self, message: str):
+        self.global_error_state.errors.append(inspect.cleandoc(message))
+
+    def is_error_state(self):
```

Comment:
did you mean 'is_in_error_state'?

Reply:
Well, I think it is fine either way.
However, if 'is_in_error_state' seems better to you, I will fix it.

Reply:
I think 'is_in_error_state' is more clear